test/unit/deepMerge.test.ts

```diff
@@ -12,19 +12,14 @@ describe("deepMerge", () => {
         expect(deepMerge({ foo: "bar" }, [2, 3, 4], true)).toEqual([2, 3, 4]);
     });
 
-    it("concatenates arrays", () => {
-        expect(deepMerge([1, 2, 3], [4, 5, 6], true)).toEqual([1, 2, 3, 4, 5, 6]);
+    it("intersects arrays", () => {
+        expect(deepMerge([1, 2, 3], [4, 5, 6], true)).toEqual([]);
         expect(deepMerge([1, 2, 3], [1, 2, 3], true)).toEqual([1, 2, 3]);
-        expect(deepMerge([1, [2], 3], [1, [2], 3], true)).toEqual([1, [2], 3]);
-        expect(deepMerge([1, { foo: "bar" }, 3], [1, { bar: "foo" }, 3], true)).toEqual([
-            1,
-            { foo: "bar" },
-            3,
-            { bar: "foo" },
-        ]);
+        expect(deepMerge([1, 2, 3], [4, 2, 6], true)).toEqual([2]);
+        expect(deepMerge([1, { foo: "bar" }], [1, { bar: "foo" }, 3], true)).toEqual([1]);
```

Comment:
Add another test case. This fails right now but I think it should work, no?

```suggestion
        expect(deepMerge([1, { foo: "bar" }], [1, { foo: "bar" }, 3], true)).toEqual([1, { foo: "bar" }]);
        expect(deepMerge([1, { foo: "bar" }], [1, { bar: "foo" }, 3], true)).toEqual([1]);
```